src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridSmtpTransport.php

```diff
@@ -20,9 +20,9 @@
  */
 class SendgridSmtpTransport extends EsmtpTransport
 {
-    public function __construct(#[\SensitiveParameter] string $key, ?EventDispatcherInterface $dispatcher = null, ?LoggerInterface $logger = null)
+    public function __construct(#[\SensitiveParameter] string $key, ?EventDispatcherInterface $dispatcher = null, ?LoggerInterface $logger = null, private ?string $region = null)
     {
-        parent::__construct('smtp.sendgrid.net', 465, true, $dispatcher, $logger);
+        parent::__construct('null' !== $region ? \sprintf('smtp.%s.sendgrid.net', $region) : 'smtp.sendgrid.net', 465, true, $dispatcher, $logger);
```

Comment:
nitpick: Just for consistency, shouldn't you use either str_replace or sprintf everywhere ?

Reply:
here, using `sprintf` will be faster than `str_replace` on PHP 8.4 (thanks to the optimization).

But we cannot use `sprintf` in the other transport without changing entirely how it is implemented (`str_replace` and `sprintf` are not replacements of each other)